src/Microsoft.AspNet.Identity/TaskExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Identity
+{
+    public static class TaskExtensions
+    {
+        public static async Task<IdentityResult> WithLoggingAsync<TUser>(this Task<IdentityResult> identityResult, UserManager<TUser> userManager,TUser user,
+            [System.Runtime.CompilerServices.CallerMemberName] string methodName = "") where TUser :class
+        {
+            return await (await identityResult).WithLoggingAsync(userManager, user, methodName);
```

Comment:
sure that is one place I need feedback on. Definitely felt wrong adding that. Alternatively it would be `var result = await task` and then call the method on it which is the same thing.

One thing I looked into was `Task.ContinueWith` but the parameter was `Task<IdentityResult>` and not `IdentityResult` so the await had to be there anyway